iroh-net/src/magic_endpoint.rs

```diff
@@ -432,153 +432,172 @@ pub async fn get_peer_id(connection: &quinn::Connection) -> anyhow::Result<PeerI
-//         assert_eq!(res.unwrap_err(), expected_err);
-
-//         accept.await.unwrap();
-//         cleanup().await;
-//     }
-
-//     #[tokio::test]
-//     async fn magic_endpoint_bidi_send_recv() {
-//         setup_logging();
-//         let (ep1, ep2, cleanup) = setup_pair().await.unwrap();
-
-//         let peer_id_1 = ep1.peer_id();
-//         eprintln!("peer id 1 {peer_id_1}");
-//         let peer_id_2 = ep2.peer_id();
-//         eprintln!("peer id 2 {peer_id_2}");
-
-//         let endpoint = ep2.clone();
-//         let p2_connect = tokio::spawn(async move {
-//             let conn = endpoint.connect(peer_id_1, TEST_ALPN, &[]).await.unwrap();
-//             let (mut send, mut recv) = conn.open_bi().await.unwrap();
-//             send.write_all(b"hello").await.unwrap();
-//             send.finish().await.unwrap();
-//             let m = recv.read_to_end(100).await.unwrap();
-//             assert_eq!(&m, b"world");
-//         });
-
-//         let endpoint = ep1.clone();
-//         let p1_accept = tokio::spawn(async move {
-//             let conn = endpoint.accept().await.unwrap();
-//             let (peer_id, alpn, conn) = accept_conn(conn).await.unwrap();
-//             assert_eq!(peer_id, peer_id_2);
-//             assert_eq!(alpn.as_bytes(), TEST_ALPN);
-
-//             let (mut send, mut recv) = conn.accept_bi().await.unwrap();
-//             let m = recv.read_to_end(100).await.unwrap();
-//             assert_eq!(m, b"hello");
-
-//             send.write_all(b"world").await.unwrap();
-//             send.finish().await.unwrap();
-//         });
-
-//         let endpoint = ep1.clone();
-//         let p1_connect = tokio::spawn(async move {
-//             let conn = endpoint.connect(peer_id_2, TEST_ALPN, &[]).await.unwrap();
-//             let (mut send, mut recv) = conn.open_bi().await.unwrap();
-//             send.write_all(b"ola").await.unwrap();
-//             send.finish().await.unwrap();
-//             let m = recv.read_to_end(100).await.unwrap();
-//             assert_eq!(&m, b"mundo");
-//         });
-
-//         let endpoint = ep2.clone();
-//         let p2_accept = tokio::spawn(async move {
-//             let conn = endpoint.accept().await.unwrap();
-//             let (peer_id, alpn, conn) = accept_conn(conn).await.unwrap();
-//             assert_eq!(peer_id, peer_id_1);
-//             assert_eq!(alpn.as_bytes(), TEST_ALPN);
-
-//             let (mut send, mut recv) = conn.accept_bi().await.unwrap();
-//             let m = recv.read_to_end(100).await.unwrap();
-//             assert_eq!(m, b"ola");
-
-//             send.write_all(b"mundo").await.unwrap();
-//             send.finish().await.unwrap();
-//         });
-
-//         p1_accept.await.unwrap();
-//         p2_connect.await.unwrap();
-
-//         p2_accept.await.unwrap();
-//         p1_connect.await.unwrap();
-
-//         cleanup().await;
-//     }
-// }
+#[cfg(test)]
+mod tests {
+    // use futures::future::BoxFuture;
```

Comment:
remove?

Reply:
it's part of the still-commented-out other test.  but sure I could remove it.